### PR TITLE
Inference support for windows

### PR DESCRIPTION
**Pull request by gnhdnb**
Tried to abstract `torch.distributed` the most subtle way possible, so all references from the jukebox are replaced with the adapter:
- If `torch.dist` is available, the adapter proxies the call further into the framework.
- If `torch.dist` is not available, the adapter emulates single node behavior.

I haven't modified any references from the APEX as it should support windows by default.

Training is not supported for windows as it is relying a lot on the distributed processing (`DistributedDataParallel`, `DistributedSampler`, fork multiprocessing, etc)

The models has to be downloaded manually at the moment for windows, I'll fix it in the next PR.

Also, few modifications has been made:
- `mpi4py` is installed from pip instead of conda (`mpi4py` is not available from conda for windows for some reason)
- explcit `encoding="utf-8"` is provided when opening the genre/artist labels in `artist_genre_processor.py`

Installation has been tested on clean environments for windows and ubuntu 

Inference has been tested for:
- 1b_lyrics, windows, single node without torch.distributed available
- 1b_lyrics, ubuntu, single node with torch.distributed available

Training has been tested for:
- small_vqvae, ubuntu, single node with torch.distributed available

Issue: https://github.com/openai/jukebox/issues/36

### CODE REVIEW

File: jukebox/utils/dist_adapter.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    BOR = 5, # Bitwise OR
+    BXOR = 6, # Bitwise XOR
+
+    def ToDistOp(self):
+        return {
+            self.SUM: dist.ReduceOp.SUM,
+            self.PRODUCT: dist.ReduceOp.PRODUCT,
+            self.MIN: dist.ReduceOp.MIN,
+            self.MAX: dist.ReduceOp.MAX,
+            self.BAND: dist.ReduceOp.BAND,
```
Comment (heewooj) [inline jukebox/utils/dist_adapter.py:19]: nitpick: We provide an option to use fused_adam with pytorch 1.1.0 and apex, but bitwise ops are not supported in 1.1.0. Since these ops are not used in this repo, could we remove them for now?

**Comment by heewooj** [comment]
Thanks for adding this! LGTM modulo a minor nitpick

File: jukebox/utils/dist_adapter.py (hunk 1 above)
Comment (gnhdnb) [reply jukebox/utils/dist_adapter.py]: sure thing
